script.js

```diff
@@ -1,53 +1,101 @@
+    return CURRENT_USAGE + fileSize <= TOTAL_DISK_SPACE;
+  }
+
+  const storeFile = (file) => {
+    CURRENT_USAGE += file.size;
+    localStorage[CACHE_CURRENT_USAGE] = CURRENT_USAGE;
+  }
+
+  const fileHandles = await window.showOpenFilePicker(PICKER_OPTS);
 
-        if (!filePattern.test(file.type)) {
-            alert("File format isn't supported");
-            return;
-        }
+  for (const fileHandle of fileHandles) {
+    const file = await fileHandle.getFile();
 
-        if (currentUsage + file.size <= totalDiskSpace) {
-            currentUsage += file.size;
-            localStorage['currentUsage'] = currentUsage;
-            updateUsageDomElements();
-        } else {
-            alert('There is not enough space on the disk');
-        }
+    if (!isFileFormatValid(file.type)) {
+      alert("File format isn't supported");
+      return;
     }
+
+    if (isThereEnoughStorageLeft(file.size)) {
+      storeFile(file);
+      updateUsageDomElements();
+    } else {
+      alert('There is not enough space on the disk');
+    }
+  }
 }
 
-document.addEventListener('DOMContentLoaded', function () {
-    document.getElementById('mb-limit').innerHTML = bytesToMbs(totalDiskSpace) + mbText;
-    updateUsageDomElements();
-});
+const updateUsageDomElements = () => {
+  const updateStorageLeftElement = () => {
+    const prettifiedStorageLeft = getPrettifiedFileSize(TOTAL_DISK_SPACE - CURRENT_USAGE);
+    document.getElementById('storage-left').innerHTML = prettifiedStorageLeft.size + '&nbsp;<span>' + prettifiedStorageLeft.units + '&nbsp;left</span> ';
+  }
+
+  const updateStorageUsageElement = () => {
+    document.getElementById('storage-usage').innerHTML = getSizeString(CURRENT_USAGE);
+  }
+
+  const updateBarElement = () => {
+    const getUsagePercentage = () => {
+      return (Math.round((CURRENT_USAGE / TOTAL_DISK_SPACE) * 100)) + '%';
+    }
 
-function updateUsageDomElements() {
-    document.getElementById('mb-usage').innerHTML = bytesToMbs(currentUsage) + mbText;
-    document.getElementById('mb-left').innerHTML = bytesToMbs(totalDiskSpace - currentUsage) + '&nbsp;<span>MB left</span> ';
     document.getElementsByClassName('gradient-bar')[0].style.width = getUsagePercentage();
+  }
+
+  updateStorageUsageElement();
+  updateStorageLeftElement();
+  updateBarElement();
 }
 
-function getUsagePercentage() {
-    return (Math.round((currentUsage / totalDiskSpace) * 100)) + '%';
+const getSizeString = (sizeInBytes) => {
+  const prettifiedSize = getPrettifiedFileSize(sizeInBytes);
+  return prettifiedSize.size + prettifiedSize.units;
 }
 
-function bytesToMbs(size) {
-    return Math.round(size / mbSizeInBytes);
+const getPrettifiedFileSize = (sizeInBytes) => {
+  let i = -1;
+  let byteUnits = [' kB', ' MB', ' GB', ' TB', 'PB', 'EB', 'ZB', 'YB'];
+
+  do {
+    sizeInBytes /= 1024;
+    i++;
+  } while (sizeInBytes > 1024);
+
+  return { size: Math.max(sizeInBytes, 0.1).toFixed(0), units: byteUnits[i] };
```

Comment:
A bit wierd that you start the index at -1 (Not code wise, we usualy start loops at 0 index logicaaly), do you think something like this will work : 

```js
 let i = 0;
  let byteUnits = [' kB', ' MB', ' GB', ' TB', 'PB', 'EB', 'ZB', 'YB'];

  do {
    sizeInBytes /= 1024;
    i++;
  } while (sizeInBytes >= 1024);

```